### PR TITLE
Fix all remaining phpstan issues

### PR DESCRIPTION
**Pull request by Davidmattei**
|Q              |A  |
|---------------|---|
|Bug fix?       |n|
|New feature?   |n|	
|BC breaks?     |n|	
|Deprecations?  |n|	
|Fixed tickets  |n|	

Fix all remaining phpstan issues

### CODE REVIEW

File: src/Helper/Api/ApiService.php [hunk 1]
```diff
@@ -110,23 +111,7 @@ private function createContentFileHashField(string $apiName, UploadedFile $file)
...
     }
 
-    public function addClientRequest(ClientRequest $clientRequest)
-    {
-        $name = $clientRequest->getOption('[api][name]', false);
-
-        if ($name) {
-            $this->clientRequests[$name] = $clientRequest;
-        }
-    }
```
Comment (theus77) [inline src/Helper/Api/ApiService.php:120]: It's used the core bundle. Please run the phpstan in the core as well

File: src/Helper/Request/ExceptionHelper.php [hunk 2]
```diff
@@ -60,12 +43,9 @@ public function renderError(FlattenException $exception)
...
     }
 
-    /**
-     * @return string
-     */
-    private function getTemplate(string $code)
+    private function getTemplate(int $code): string
     {
-        $customCodeTemplate = \str_replace('{code}', $code, $this->template);
+        $customCodeTemplate = \str_replace('{code}', (string) $code, $this->template);
```
Comment (theus77) [inline src/Helper/Request/ExceptionHelper.php:48]: Prefer \strval

File: src/Helper/Request/LocaleHelper.php [hunk 3]
```diff
@@ -1,33 +1,29 @@
...
-class LocaleHelper
+final class LocaleHelper
 {
-    /**
-     * @var RouterInterface
-     */
-    private $router;
+    private RouterInterface $router;
+    /** @var string[] */
+    private array$locales;
```
Comment (theus77) [inline src/Helper/Request/LocaleHelper.php:15]: Missing space

File: src/Helper/Routing/Route.php [hunk 4]
```diff
@@ -1,32 +1,39 @@
...
+     * @param string[] $locales
+     */
     public function addToCollection(RouteCollection $collection, array $locales = []): void
     {
         $path = $this->options['path'];
 
         if (\is_array($path)) {
             foreach ($path as $key => $p) {
-                $locale = \in_array($key, $locales) ? $key : null;
+                $locale = \in_array($key, $locales) ? (string) $key : null;
```
Comment (theus77) [inline src/Helper/Routing/Route.php:36]: Strval ?

File: src/Helper/Routing/Router.php [hunk 5]
```diff
@@ -140,7 +148,7 @@ private function getRoutesByEnvironment(ClientRequest $clientRequest, Environmen
             $clientRequest->cacheContentType($contentType);
 
             return $routes;
-        } catch (EnvironmentNotFoundException $e) {
+        } catch (\Exception $e) {
```
Comment (theus77) [inline src/Helper/Routing/Router.php:151]: Prefer Throwable

File: src/Helper/Twig/TwigLoader.php [hunk 6]
```diff
@@ -33,33 +33,23 @@ public function __construct(ClientRequest $client, array $config)
...
-    {
-        return new \Twig_Source($this->getTemplateCode($name), $name);
-    }
-
-    /**
-     * {@inheritdoc}
-     *
-     * @deprecated used for php < 7
-     */
-    public function getSource($name)
```
Comment (theus77) [inline src/Helper/Twig/TwigLoader.php:46]: Delete?

**Review by theus77 — COMMENTED** [review]
I think that core's needs have been dropped

File: src/Helper/Twig/TwigLoader.php (hunk 6 above)
Comment (Davidmattei) [reply src/Helper/Twig/TwigLoader.php:46]: Yes was only needed in version before php7

File: src/Helper/Routing/Router.php (hunk 5 above)
Comment (Davidmattei) [reply src/Helper/Routing/Router.php]: Good remark! fix in 313a138

File: src/Helper/Routing/Route.php (hunk 4 above)
Comment (Davidmattei) [reply src/Helper/Routing/Route.php]: 313a138

File: src/Helper/Request/LocaleHelper.php (hunk 3 above)
Comment (Davidmattei) [reply src/Helper/Request/LocaleHelper.php]: Pro review! 313a138

File: src/Helper/Request/ExceptionHelper.php (hunk 2 above)
Comment (Davidmattei) [reply src/Helper/Request/ExceptionHelper.php]: 313a138

File: src/Controller/ApiController.php [hunk 7]
```diff
@@ -10,56 +12,35 @@
...
-     * @param string $contentType
-     *
-     * @return JsonResponse
-     */
-    public function contentType(Request $request, $apiName, $contentType)
+    public function contentType(Request $request, string $apiName, string $contentType): JsonResponse
     {
         $scrollId = $request->query->get('scroll');
-        $size = $request->query->get('size');
+        $size = (int) $request->query->get('size');
```
[the remark this answers is not in the document]
Comment (Davidmattei) [reply src/Controller/ApiController.php]: 313a138

File: src/Command/HealthCheckCommand.php [hunk 8]
```diff
@@ -57,23 +57,16 @@ protected function execute(InputInterface $input, OutputInterface $output): int
         $io = new SymfonyStyle($input, $output);
         $io->title('Performing Health Check');
 
-        $this->checkElasticSearch($io, $input->getOption('green'));
+        $this->checkElasticSearch($io, (bool) $input->getOption('green'));
```
[the remark this answers is not in the document]
Comment (Davidmattei) [reply src/Command/HealthCheckCommand.php]: 313a138

File: src/Helper/Api/ApiService.php (hunk 1 above)
Comment (Davidmattei) [reply src/Helper/Api/ApiService.php:120]: Check the full coreBundle but addClientRequest is not called, this is on hte apiService helper. The function was in the first version of the ApiService, after the clientRequests were injected in the controller and we forgot to remove this function.

Comment (Davidmattei) [reply src/Helper/Api/ApiService.php:120]: The core only uses the Environment and EnvironmentHelper class, plus it has custom route for ems_core_asset_proxy which goes to EMS\ClientHelperBundle\Controller\AssetController:proxyFromRefererPattern. 

In a next version/pr I will implement contract interfaces in the clientHelperBundle and the core and formBundle will only use interfaces of the clientHelper. That way we see when we break something

**Review by theus77 — APPROVED** [review]

**Pull request merged**